src/components/CheckIn/CheckIn.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import Route from 'react-router-dom/Route';
+
+import {
+  Button,
+  ButtonGroup,
+  Col,
+  IconButton,
+  Pane,
+  PaneMenu,
+  Paneset,
+  Row,
+} from '@folio/stripes/components';
+import { stripesShape } from '@folio/stripes/core';
+
+import { ORDER } from '../Utils/resources';
+import CheckInItems from './CheckInItems';
+import CheckInHistory from './CheckInHistory';
+
+export const ITEMS = '/items';
+export const HISTORY = '/history';
+
+class CheckIn extends Component {
+  static manifest = Object.freeze({
+    query: {},
+    order: ORDER,
+  });
+
+  static propTypes = {
+    location: ReactRouterPropTypes.location.isRequired,
+    match: ReactRouterPropTypes.match,
+    mutator: PropTypes.object.isRequired,
+    resources: PropTypes.object,
+    stripes: stripesShape.isRequired,
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.connectedCheckInItems = props.stripes.connect(CheckInItems);
+    this.connectedCheckInHistory = props.stripes.connect(CheckInHistory);
+  }
+
+  getButtonStyle = (pathEnding) => {
+    const { location } = this.props;
+
+    return location.pathname.endsWith(pathEnding)
+      ? 'primary'
+      : 'default';
+  };
+
+  onClose = () => {
+    const { location, mutator } = this.props;
+    const pathEnding = location.pathname.endsWith(ITEMS)
+      ? ITEMS
+      : HISTORY;
+
+    mutator.query.replace({
+      _path: location.pathname.replace(`/check-in${pathEnding}`, ''),
+    });
+  }
+
+  getFirstMenu = () => (
+    <PaneMenu>
+      <FormattedMessage id="ui-orders.buttons.line.close">
+        {(title) => (
+          <IconButton
+            ariaLabel={title}
+            data-test-close-button
+            icon="times"
+            onClick={this.onClose}
+          />
+        )}
+      </FormattedMessage>
+    </PaneMenu>
+  );
+
+  render() {
+    const { match, resources, stripes } = this.props;
+    const orderNumber = get(resources, ['order', 'records', 0, 'poNumber'], '');
+
+    return (
+      <Paneset>
+        <Pane
+          defaultWidth="fill"
+          paneTitle={(
+            <FormattedMessage
+              id="ui-orders.checkIn.paneTitle"
+              values={{ orderNumber }}
+            />
+
+          )}
+          firstMenu={this.getFirstMenu()}
+        >
+          <Row center="xs">
+            <Col xs>
+              <ButtonGroup>
```

Comment:
I'd suggest to move it to separate component to have `tiny` container

Reply:
I think it will require a lot more code, or I didn't get you right

Reply:
I don't know how you get it :)

not a lot, few lines, just create a component `CheckInNavigation` where you can move all related to it code and not to keep it in root component

Reply:
fixed